pdns/logger.cc

```diff
@@ -47,17 +47,20 @@ void Logger::log(const string &msg, Urgency u)
 #ifndef RECURSOR
   bool mustAccount(false);
 #endif
-  struct tm tm;
-  time_t t;
-  time(&t);
-  tm=*localtime(&t);
-
   if(u<=consoleUrgency) {
-    char buffer[50];
-    strftime(buffer,sizeof(buffer),"%b %d %H:%M:%S ", &tm);
+    if (d_timestamps) {
+      char buffer[50] = "";
+      struct tm tm;
+      time_t t;
+      time(&t);
+      tm=*localtime(&t);
+      strftime(buffer,sizeof(buffer),"%b %d %H:%M:%S ", &tm);
+      clog << string(buffer);
```

Comment:
This write is unlocked, perhaps it would be easier to declare buffer out of the `if (d_timestamps) {` block, initialize it to an empty string like it's done in your code, and keep the existing `clog << string(buffer) + msg <<endl;`?

Reply:
nice catch, I was hoping to avoid initializing the buffer for every log line if unneeded, maybe already lock before `if (d_timestamps)`?

Reply:
That would mean calling `localtime()` and `strftime()` while holding the lock, which we should try very hard to avoid. A 50 bytes buffer on the stack is cheap and initializing it to "" is just setting the first byte to `\0`, so IMHO that's a better choice.

Reply:
ack